framework/db/pgsql/ArrayParser.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+namespace yii\db\pgsql;
+
+/**
+ * TODO: phpdoc, tests
+ *
+ * @author Sergei Tigrov <rrr-r@ya.ru>
+ * @author Dmytro Naumenko <d.naumenko.a@gmail.com>
+ */
+class ArrayParser
+{
+    /**
+     * @var string Character used in array
+     */
+    protected $delimiter = ',';
+
+    /**
+     * Convert array from PostgreSQL to PHP
+     *
+     * @param string $value string to be converted
+     * @return array|null
+     */
+    public function parse($value)
+    {
+        if (is_null($value)) {
+            return null;
+        }
+
+        if ($value === '{}') {
+            return [];
+        }
+
+        return $this->parseArray($value);
+    }
+
+    private function parseArray($value, &$i = 0)
+    {
+        $result = [];
+        for(++$i; $i < strlen($value); ++$i) {
```

Comment:
1. `$value` could be in UTF so `strlen` won't work.
2. Formatting.
3. `strlen` should not be called in condition part of `for`. Either in pre-condition or out of cycle.